docs/providers/aws/guide/plugins.md

```diff
@@ -510,6 +510,55 @@ This way, if user sets `anotherProp` by mistake to `some-string`, the Framework
 Serverless: Configuration error: functions.someFunc.events[0].yourPluginEvent.anotherProp should be number
 ```
 
+#### `defineFunctionProperties` helper
+
+Let's say your plugin adds support to a new `someProperty` function property. To use this property, a user would need to have `serverless.yaml` file like this:
+
+```
+// serverless.yaml
+
+functions:
+  someFunc:
+    handler: handler.main
+    someProperty: my-property-value
+```
+
+In this case your plugin should add validation rules inside your plugin constructor. Otherwise, the Framework would display an error message saying that your property is not supported:
+
+```
+ServerlessError: Configuration error:
+at 'functions.someFunc': unrecognized property 'someProperty'
+```
+
+To fix this error and more importantly to provide validation rules for your property, modify your plugin constructor with code like this:
+
+```javascript
+class NewEventPlugin {
+  constructor(serverless) {
+    this.serverless = serverless;
+
+    // Create schema for your properties. For reference use https://github.com/ajv-validator/ajv
+    serverless.configSchemaHandler.defineFunctionProperties('providerName', {
+      type: 'object',
+      properties: {
+        someProperty: { type: 'string' },
+        anotherProperty: { type: 'number' },
+      },
+      required: ['someProperty'],
+      additionalProperties: false,
+    });
+  }
+}
+```
+
+This way, if user sets `anotherProperty` by mistake to `hello`, the Framework would display an error:
+
+```
+ServerlessError: Configuration error at 'functions.someFunc.anotherProperty': should be number
+```
+
+Note, however, that if the user sets `someProperty` to a number, it will just be treated as a string.
```

Comment:
Let's remove this note. This refers to general coercion rules which are not specific to this helper

Reply:
Ok, removed